lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/cache/TestEvictableLoadingCache.java

```diff
@@ -0,0 +1,293 @@
+import io.airlift.concurrent.MoreFutures;
+import org.gaul.modernizer_maven_annotations.SuppressModernizer;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.OptionalLong;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.IntStream;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.trino.plugin.base.cache.CacheStatsAssertions.assertCacheStats;
+import static java.lang.String.format;
+import static java.util.concurrent.Executors.newFixedThreadPool;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotSame;
+import static org.testng.Assert.assertSame;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+public class TestEvictableLoadingCache
+{
+    private static final int TEST_TIMEOUT_MILLIS = 10_000;
+
+    @Test(timeOut = TEST_TIMEOUT_MILLIS)
+    public void testLoad()
+            throws Exception
+    {
+        LoadingCache<Integer, String> cache = EvictableLoadingCache.build(
+                OptionalLong.empty(),
+                OptionalLong.empty(),
+                10_000,
+                true,
+                CacheLoader.from((Integer ignored) -> "abc"));
+
+        assertEquals(cache.get(42), "abc");
+    }
+
+    @Test(timeOut = TEST_TIMEOUT_MILLIS)
+    public void testLoadStats()
+            throws Exception
+    {
+        LoadingCache<Integer, String> cache = EvictableLoadingCache.build(
+                OptionalLong.empty(),
+                OptionalLong.empty(),
+                10_000,
+                true,
+                CacheLoader.from((Integer ignored) -> "abc"));
+
+        assertEquals(cache.stats(), new CacheStats(0, 0, 0, 0, 0, 0));
+
+        String value = assertCacheStats(cache)
+                .misses(1)
+                .loads(1)
+                .calling(() -> cache.get(42));
+        assertEquals(value, "abc");
+
+        value = assertCacheStats(cache)
+                .hits(1)
+                .calling(() -> cache.get(42));
+        assertEquals(value, "abc");
+
+        // with equal, but not the same key
+        value = assertCacheStats(cache)
+                .hits(1)
+                .calling(() -> cache.get(newInteger(42)));
+        assertEquals(value, "abc");
+    }
+
+    @SuppressModernizer
+    private static Integer newInteger(int value)
```

Comment:
I would prefer a dedicated class. It would be known what is the goal of it. No need to copy paste, no need for `SuppressWarnings`. Class could have a javadoc etc.

Reply:
Just boiler plate. Suppressing something is cheaper that writing a class.